spec/facter/facts/debian/os/architecture_spec.rb

```diff
@@ -4,61 +4,16 @@
   describe '#call_the_resolver' do
     subject(:fact) { Facts::Debian::Os::Architecture.new }
 
-    context 'when resolver does not return x86_64' do
-      let(:value) { 'i86pc' }
+    it 'calls Facter::Core::Execution and returns architecture fact' do
+      %w[amd64 arm64 i386].each do |arch|
```

Comment:
Isn't it better to swap the order? So do the each and then `it`, making them present as separate examples.